enclave/core/sgx/enter.S

```diff
@@ -105,15 +105,36 @@ oe_enter:
 
     // Abort if SSA[0].GPRSGX.URSP is within the enclave memory range
     cmp %r12, %r9
-    jb .exception_check_pass
+    jb .exception_handler_stack_check
     cmp %r13, %r9
-    jae .exception_check_pass
+    jae .exception_handler_stack_check
     jmp .abort
 
     // Reaching this point implies SSA[0].GPRSGX.RSP is within the enclave
     // memory range so we do not need additional checks.
 
-.exception_check_pass:
+.exception_handler_stack_check:
+    // Stop speculative execution at target of conditional jump
+    lfence
+
+    // Get the exception_handler_stack_check range
+    mov td_exception_handler_stack(%r11), %r14
+    mov td_exception_handler_stack_size(%r11), %r15
+    add %r14, %r15
+    jc .exception_stack_setup // check for overflow
+
+    // Check if the stack range is within the enclave memory range
+    // If the check fails, fallback to the default behavior (i.e.,
+    // re-using the stack pointer saved in the SSA)
+    cmp %r12, %r14
+    jb .exception_stack_setup
+    cmp %r13, %r15
+    ja .exception_stack_setup
+
+    // Check passes, use the exception handler stack
+    mov %r15, %r8
```

Comment:
ABI_REDZONE_BYTES at the top of the exception stack are never used.

Reply:
This does not seem to be addressed.

Reply:
Now after all the checks pass, the code directly jumps to `.call_function` without subtracting ABI_REDZONE_BYTES from stack pointer. Is this what's you suggested?

Reply:
Can you double-check if we need to add an lfence at the fall-through of the exception handler conditional jumps?